src/app/core/facades/account.facade.ts

```diff
@@ -251,6 +253,16 @@ export class AccountFacade {
     this.store.dispatch(updateCustomerAddress({ address }));
   }
 
+  // NEWSLETTER
+
+  subscribedToNewsletter$ = this.store.pipe(select(getNewsletterSubscriptionStatus));
+
+  updateNewsletterSubscription(subscribedToNewsletter: boolean) {
```

Comment:
here the configuration parameter should be checked as well before the update action is called